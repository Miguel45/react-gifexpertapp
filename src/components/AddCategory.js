import React from 'react'
import PropTypes from 'prop-types';

const defaultValue = '';

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = React.useState(defaultValue)
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        if(inputValue.trim().length > 2){
            setCategories(cats => [inputValue, ...cats])
            setInputValue(defaultValue)
        }
    }

    return (
        <form onSubmit={ handleSubmit } >
            <input
            type='text'
            value={inputValue}
            onChange={handleInputChange}/>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
