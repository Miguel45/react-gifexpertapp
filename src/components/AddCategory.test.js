import React from 'react' 
import { AddCategory } from './AddCategory'
import { shallow } from 'enzyme'
import '@testing-library/jest-dom'
describe( 'Pruebas en el archivo AddCategory.js', () => {

    const setCategories = jest.fn()
    let wrapper = shallow( <AddCategory setCategories={setCategories}/> );

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    })

    //Enzyme
    test( 'Debe mostrarse bien el componente en un snapshot', () => {

        expect( wrapper ).toMatchSnapshot();

    })

    test('debe de cambiar la caja de texto', () => {
        
        const value = 'hola mundo';
        
        wrapper.find('input').simulate('change', { target: { value } });

        expect( wrapper.find('input').prop('value') ).toBe(value);

    })

    test('no debe de postear la informaciÃ³n on submit', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault: () => {} });
        
        expect( setCategories ).not.toHaveBeenCalledWith( expect.any( Function ) );

    })

    test('debe de llamar el setCategories y limpiar la caja de texto', () => {

        const value = 'hola mundo';

        wrapper.find('input').simulate('change', { target: { value } });

        wrapper.find('form').simulate('submit', { preventDefault: () => {} });

        expect( setCategories ).toHaveBeenCalled(  );
        expect( setCategories ).toHaveBeenCalledTimes( 1 );
        expect( setCategories ).toHaveBeenCalledWith( expect.any( Function ) );

        expect( wrapper.find('input').prop('value') ).toBe('');

    })

})