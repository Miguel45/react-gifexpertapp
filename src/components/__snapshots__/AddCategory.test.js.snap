// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo AddCategory.js Debe mostrarse bien el componente en un snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
